syntax = "proto3";

package transporter;

message Climate {
  uint32 id = 1;
  uint32 dht22_port = 2;
  uint32 aqi_port = 3;
  bool has_buzzer = 4;
  uint32 buzzer_port = 5;
}

message Ldr {
  uint32 id = 1;
  uint32 port = 2;
}

message Motion {
  uint32 id = 1;
  uint32 port = 2;
  uint32 relay_channel = 3;
}

message Relay {
  uint32 id = 1;
  int32 type = 2;
  uint32 lower_port = 3;
  uint32 upper_port = 4;
}

message ConfigData {
  uint32 version = 1;
  uint32 size = 2;
  uint32 climate_size = 3;
  uint32 ldr_size = 4;
  uint32 motion_size = 5;
  uint32 relay_size = 6;

  repeated Climate climates = 7; // Climate config array
  repeated Ldr ldrs = 8; // LDR config array
  repeated Motion motions = 9; // Motion sensor config array
  repeated Relay relays = 10; // Relay config array
}

message ClimateSensorData{
  uint32 id = 1;
  string device_id = 2;
  float temperature = 2;
  float humidity = 3;
  float aqi = 4;
}
